crates/matrix-sdk-crypto/src/types/events/utd_cause.rs

```diff
@@ -163,6 +190,44 @@ impl UtdCause {
             _ => UtdCause::Unknown,
         }
     }
+
+    /**
+     * Below is the flow chart we follow for deciding whether historical
+     * UTDs are expected. This function starts at position `B`.
+     *
+     * ```mermaid
+     * flowchart TD
+     * A{Is the message newer than the device?}
+     * A -- No --> B
+     * A -- Yes --> X
+     * B{Is there a backup on the server?}
+     * B -- No --> Y
+     * B -- Yes --> C
+     * C{Is backup working on this device?}
+     * C -- No --> D
+     * C -- Yes --> X
+     * D{Is this device verified?}
+     * D -- No --> Z
+     * D -- Yes --> X
+     *
+     * X([Normal UTD error])
+     * Y([History is not available on this device])
+     * Z([You need to verify this device])
+     * ```
+     */
+    fn determine_historical(crypto_context_info: CryptoContextInfo) -> UtdCause {
+        if crypto_context_info.backup_exists_on_server {
+            if crypto_context_info.is_backup_configured
+                || crypto_context_info.this_device_is_verified
+            {
+                UtdCause::Unknown
```

Comment:
A comment might help clear up the confusion here.

```suggestion
                // Either backup download is working but we still don't have the key, or backup download
                // *isn't* working despite being verified. In either case, we shrug and give an `Unknown` cause.
                UtdCause::Unknown
```